src/libopenrave/robotconnectedbody.cpp

```diff
@@ -108,52 +108,53 @@ void RobotBase::ConnectedBodyInfo::DeserializeJSON(const rapidjson::Value &value
 {
     openravejson::LoadJsonValueByKey(value, "name", _name);
     openravejson::LoadJsonValueByKey(value, "linkName", _linkname);
-    openravejson::LoadJsonValueByKey(value, "url", _url);
+    openravejson::LoadJsonValueByKey(value, "uri", _uri);
     openravejson::LoadJsonValueByKey(value, "transform", _trelative);
 
-    if(value.HasMember("links"))
```

Comment:
@ziyan @bucketzxm shouldn't you always clear `_vLinkInfos` and other structs when deserializing since they can contain old data? Why preserve them?

Reply:
This one is fixed. @bucketzxm please check other info structure and see if similar problem exists.